zvariant/src/lib.rs

```diff
@@ -1293,6 +1293,354 @@ mod tests {
         assert_eq!(dict1, dict2);
     }
 
+    fn mock_fd<'a>(value: i32) -> Fd<'a> {
+        use std::os::fd::BorrowedFd;
+        use std::os::fd::RawFd;
+        let fd = unsafe { BorrowedFd::borrow_raw(value as RawFd) };
+        Fd::from(fd)
+    }
+
+    macro_rules! dict_iter_tests_gen {
+        ($($name:ident: ($k_ty:ty, $v_ty:ty, $values:expr),)*) => {
+            $(
+                #[test]
+                fn $name() {
+                    let mut dict = Dict::new(<$k_ty>::signature(), <$v_ty>::signature());
+                    for (k, v) in $values {
+                        dict.add(k, v).unwrap();
+                    }
+                    let mut expect = Vec::<(Value<'_>, Value<'_>)>::new();
+                    for (k, v) in $values {
+                        expect.push((Value::from(k), Value::from(v)));
+                    }
+                    let expect_iter = expect.iter().map(|(k, v)| (k, v)).collect::<Vec<_>>();
+                    let actual = dict.iter().collect::<Vec<_>>();
+                    assert_eq!(actual, expect_iter);
+                    let actual = (&dict).iter().collect::<Vec<_>>();
+                    assert_eq!(actual, expect_iter);
+                    let actual = (&mut dict).iter().collect::<Vec<_>>();
+                    assert_eq!(actual, expect_iter);
+                    let actual = dict.into_iter().collect::<Vec<_>>();
+                    assert_eq!(actual, expect);
+                }
+            )*
+        }
+    }
```

Comment:
Now we have a macro that is not used? :thinking:

Reply:
Removed the redundant macro.